include/battle.h

```diff
@@ -718,6 +718,7 @@ struct BattleStruct
     bool8 trainerSlideBeforeFirstTurnMsgDone;
     u32 aiDelayTimer; // Counts number of frames AI takes to choose an action.
     u32 aiDelayFrames; // Number of frames it took to choose an action.
+    bool8 transformZeroToHero[PARTY_SIZE][NUM_SIDES];
```

Comment:
Whoops :D
```suggestion
    bool8 transformZeroToHero[PARTY_SIZE][NUM_BATTLE_SIDES];
```

Reply:
Forced push a commit with this change plus adds it to `allowedToChangeFormInWeather`